s2/cellunion.go

```diff
@@ -130,15 +130,48 @@ func CellUnionFromIntersectionWithCellID(x CellUnion, id CellID) CellUnion {
 	return cu
 }
 
+// cellUnionDifferenceInternal adds (xid - y) to the CellUnion. It subdivides
+// xid when there is partial overlap, narrowing y before recursing.
+func (cu *CellUnion) cellUnionDifferenceInternal(xid CellID, y CellUnion) {
```

Comment:
I moved this prior to the exported method to match C++ and have better proximity.